pkg/providers/instancetype/types.go

```diff
@@ -485,10 +485,7 @@ func systemReservedResources(systemReserved map[string]string) corev1.ResourceLi
 	})
 }
 
-func kubeReservedResources(cpus, pods, eniLimitedPods *resource.Quantity, amiFamily amifamily.AMIFamily, kubeReserved map[string]string) corev1.ResourceList {
-	if amiFamily.FeatureFlags().UsesENILimitedMemoryOverhead {
-		pods = eniLimitedPods
-	}
+func kubeReservedResources(cpus, pods *resource.Quantity, kubeReserved map[string]string) corev1.ResourceList {
```

Comment:
Can we change this function signature so that we don't pass `pods` like this -- I personally find this a little confusing because what we're passing here isn't actually `pods` but it's just the ENILimitedPods